File: go/pkg/models/point_of_interest_team.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type PointOfInterestTeam struct {
	ID                uuid.UUID       `json:"id" gorm:"type:uuid;default:uuid_generate_v4()"`
	CreatedAt         time.Time       `json:"createdAt"`
	UpdatedAt         time.Time       `json:"updatedAt"`
	TeamID            uuid.UUID       `json:"teamId"`
	Team              Team            `json:"team"`
	PointOfInterestID uuid.UUID       `json:"pointOfInterestId"`
	PointOfInterest   PointOfInterest `json:"pointOfInterest"`
	Unlocked          bool            `json:"unlocked"`
	CaptureTier       int             `json:"captureTier"`
}

File: js/packages/types/src/pointOfInterestTeam.ts
export interface PointOfInterestTeam {
    id: string;
    createdAt: Date;
    updatedAt: Date;
    teamId: string;
    pointOfInterestId: string;
    tierOneCaptured: boolean;
    tierTwoCaptured: boolean;
    tierThreeCaptured: boolean;
}
